charmhelpers/contrib/storage/linux/ceph.py

```diff
@@ -382,13 +394,50 @@ def create(self):
                                           name=self.app_name)
                 except CalledProcessError:
                     log('Could not set app name for pool {}'.format(self.name, level=WARNING))
+                nautilus_or_later = cmp_pkgrevno('ceph-common', '14.2.0') >= 0
+                if nautilus_or_later:
+                    # Ensure we set the expected pool ratio
+                    update_pool(client=self.service,
+                                pool=self.name,
+                                settings={'target_size_ratio': str(self.percent_data / 100.0)})
+                if 'pg_autoscaler' in enabled_manager_modules():
+                    try:
+                        enable_pg_autoscale(self.service, self.name)
+                    except CalledProcessError:
+                        log('Could not configure auto scaling for pool {}'.format(self.name, level=WARNING))
             except CalledProcessError:
                 raise
 
     """Get an existing erasure code profile if it already exists.
        Returns json formatted output"""
 
 
+def enabled_manager_modules():
+    """Return a list of enabled manager modules.
+
+    :returns: list[str]
+    """
+    cmd = ['ceph', 'mgr', 'module', 'ls']
+    try:
+        modules = check_output(cmd)
+    except CalledProcessError as e:
+        log("Failed to list ceph modules: {}".format(e), WARNING)
```

Comment:
For possible future debugging, would it be useful to log returncode and output from command too?

Reply:
At this point, there is no output, and the return data should be logged via the CalledProcessError?

```
out = subprocess.check_output(['/bin/sh', '-c', 'echo test && false'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/subprocess.py", line 356, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/bin/sh', '-c', 'echo test && false']' returned non-zero exit status 1.
>>> out
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'out' is not defined
>>> out = subprocess.check_output(['/bin/sh', '-c', 'echo test && true'])
>>> out
b'test\n'
```

Doing something like this snippet gives us:

```
>>> try:
...   out = subprocess.check_output(['/bin/sh', '-c', 'echo test && false'])
... except subprocess.CalledProcessError as e:
...   print("Error: {}".format(e))
... 
Error: Command '['/bin/sh', '-c', 'echo test && false']' returned non-zero exit status 1.
```

Reply:
Any output from subprocess when it returns non-zero is available in ``CalledProcessError.output``, but your change at least adds the return code which may be sufficient.